services/datcore-adapter/src/simcore_service_datcore_adapter/api/routes/datasets.py

```diff
@@ -1,38 +1,44 @@
 import logging
-import tempfile
-from pathlib import Path
-from typing import List
+from typing import Final
 
-import aiofiles
-from fastapi import APIRouter, Depends, File, Header, Request, UploadFile
+from aiocache import cached
+from fastapi import APIRouter, Depends, Header, Request
 from fastapi_pagination import Page, Params
 from fastapi_pagination.api import create_page, resolve_params
 from fastapi_pagination.bases import RawParams
+from servicelib.fastapi.requests_decorators import cancel_on_disconnect
 from starlette import status
 
 from ...models.domains.datasets import DatasetsOut, FileMetaDataOut
 from ...modules.pennsieve import PennsieveApiClient
-from ...utils.requests_decorators import cancellable_request
 from ..dependencies.pennsieve import get_pennsieve_api_client
 
 router = APIRouter()
 log = logging.getLogger(__file__)
 
+_MINUTE: Final[int] = 60
+_PENNSIEVE_CACHING_TTL_S: Final[int] = 5 * _MINUTE
+
 
 @router.get(
     "/datasets",
     summary="list datasets",
     status_code=status.HTTP_200_OK,
     response_model=Page[DatasetsOut],
 )
-@cancellable_request
+@cancel_on_disconnect
+@cached(
+    ttl=_PENNSIEVE_CACHING_TTL_S,
```

Comment:
Question, why only 5 minutes? What was the reasoning behind it. Pleas note, I don't know how this data is used usually. I think it's just displayed in the UI, right?

Reply:
Pennsieve is accessible on [https://app.pennsieve.io/](https://app.pennsieve.io/)
- you have access as well
- the data there changes over time and that is outside of our control, hence at the moment I put 5 minutes only
- one possible use-case is that a user uploads some data there, and then wants to use it in osparc, therefore I thought that more than 5 minutes might be a bit annoying